data_steward/retraction/retract_deactivated_pids.py

```diff
@@ -0,0 +1,498 @@
+                        dataset=date_row.dataset_id,
+                        table=date_row.table,
+                        pid=pid,
+                        deactivated_pids_project=pids_project_id,
+                        deactivated_pids_dataset=pids_dataset_id,
+                        deactivated_pids_table=pids_table,
+                        end_date_column=date_row.end_date_column,
+                        start_date_column=date_row.start_date_column)
+                    queries_list.append({
+                        clean_consts.QUERY: sandbox_query,
+                        clean_consts.DESTINATION_DATASET: sandbox_dataset,
+                        clean_consts.DESTINATION_TABLE: date_row.table,
+                        'type': 'sandbox'
+                    })
+                    clean_query = CLEAN_QUERY_END_DATE.render(
+                        project=date_row.project_id,
+                        dataset=date_row.dataset_id,
+                        table=date_row.table,
+                        pid=pid,
+                        deactivated_pids_project=pids_project_id,
+                        deactivated_pids_dataset=pids_dataset_id,
+                        deactivated_pids_table=pids_table,
+                        end_date_column=date_row.end_date_column,
+                        start_date_column=date_row.start_date_column)
+                    queries_list.append({
+                        clean_consts.QUERY: clean_query,
+                        clean_consts.DESTINATION_DATASET: date_row.dataset_id,
+                        clean_consts.DESTINATION_TABLE: date_row.table,
+                        clean_consts.DISPOSITION: bq_consts.WRITE_TRUNCATE,
+                        'type': 'retraction'
+                    })
+            else:
+                # break out of loop to create query, if pid does not exist in table
+                continue
+    LOGGER.info(
+        f"Query list complete, retracting ehr deactivated PIDS from the following datasets: "
+        f"{dataset_list}")
+    return queries_list
+
+
+def run_queries(queries, client):
+    """
+    Function that will perform the retraction.
+
+    :param queries: list of queries to run retraction
+    """
+    query_job_ids = []
+    for query_dict in queries:
+        if query_dict['type'] == 'sandbox':
+            LOGGER.info(
+                f"Writing rows to be retracted to, using query {query_dict['query']}"
+            )
+            response = client.query(query_dict['query'])
+            LOGGER.info(
+                f"{query_dict['destination_table_id']} table written to {query_dict['destination_dataset_id']}"
+            )
+            query_job_id = response.job_id
+            query_job_ids.append(query_job_id)
+        else:
+            LOGGER.info(
+                f"Truncating table with clean data, using query {query_dict['query']}"
+            )
+            response = client.query(query_dict['query'])
```

Comment:
You will need to add a [job_config](https://googleapis.dev/python/bigquery/latest/generated/google.cloud.bigquery.client.Client.html#google.cloud.bigquery.client.Client.query) to your query here.  Otherwise, this will select from but will not overwrite the existing table.  This is the [job_config api documentation](https://googleapis.dev/python/bigquery/latest/generated/google.cloud.bigquery.job.QueryJobConfig.html#google.cloud.bigquery.job.QueryJobConfig).  Whether you need to add one to line 409 depends on if you need to overwrite an existing table.

Reply:
This has been added, please take a look and let me know if there needs to be any changes. I ran my integration test and checked BQ and it seemed to worked appropriately.